compiler/one-cmds/one-infer

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bash
+''''export SCRIPT_PATH="$(cd "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")" && pwd)" # '''
+''''export PY_PATH=${SCRIPT_PATH}/venv/bin/python                                       # '''
+''''test -f ${PY_PATH} && exec ${PY_PATH} "$0" "$@"                                     # '''
+''''echo "Error: Virtual environment not found. Please run 'one-prepare-venv' command." # '''
+''''exit 255                                                                            # '''
+
+# Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import copy
+import glob
+import itertools
+import ntpath
+import os
+import sys
+
+import utils as _utils
+
+# TODO Find better way to suppress trackback on error
+sys.tracebacklimit = 0
+
+
+def _get_backends_list():
+    # This approach is too much backend-depended, 
```

Comment:
I'm not sure but comments sounds like to me that this approach should follow backend rules. But it is the exact opposite.
```suggestion
    # This approach forces the backend drivers to have a restricted name.
```

Reply:
Anyway, is it natural to do so? Drivers for inference seems hard to have restricted name. How about just add `-d, --driver` option where we can pass exact driver path to `one-infer`?

Reply:
As a wrapper which enables backend inference driver, it sounds better to specify its name. 

Here are some limitation if we apply the change.

1. driver candidates cannot be given to user. 
For now, `one-infer` searches `*-infer` from `one/bin/` and `one/backends/**/bin/* and shows them as backend driver candidates. `-d/--driver` option cannot suggest the same option. 

2. Cannot sure the driver argument is **inference tool**
Since `one-infer` works as a wrapper command for `one-cmds`, any driver can be passed and executed in `one-infer` stage. But, IMHO it cannot be solved in this situation. (just for note)

I'll apply the changes and revise some help messages.

Reply:
Hmmm.. on second thought, the original codes is necessary because `one-codegen` already does that. 

(optional) Maybe just adding `--driver` option will do the trick. I mean we can use both of `-b` and `-d` option. Then,

> driver candidates cannot be given to user.

Just printing `-infer` driver list is enough.

> Cannot sure the driver argument is **inference tool**

We don't need to know about this. It would cause an error.

Reply:
> Hmmm.. on second thought, the original codes is necessary because `one-codegen` already does that.
> (optional) Maybe just adding `--driver` option will do the trick. I mean we can use both of `-b` and `-d` option.

`one-codegen` does what? `-b/--backend` option? Yes, for consistency, it'd be better to remaining `-b` option.
However if there exists both `-b` and `-d` options, both of the options should be **optional** argument, not required. It's a little bit weird for me. Does it look fine?

Reply:
> Does it look fine?

Seems fine to me as of now because other tools already have `-b` option.

But, there could be some drivers that we cannot control including its name. So, we should use `-d` option later. Or, we could make `-b` option the one that receive backend driver path.

@seanshpark What do you think of this?

Reply:
> What do you think of this?

I wasn't following the context so had a short chat with @yunjayh :)
If both `-b` and `-d` exist then we can choose with `priority`; give `-d` higher so if `-d` is given we choose the driver, if not then choose `-b` option. both absent will be an error.

Reply:
Umm.. I think we don't have to receive both of them simultaneously. Just making them exclusive is enough.

Reply:
I have no preference between mutually exclusive approach and priority approach. Making them exclusive can make user behavior more clear, but defining priority level may give a chance to user for driver replacement.
How about to print the executable binary's name which is used in `one-infer` if both options are set?

Reply:
@yunjayh Seems good:) Maybe using warning would be good, either.

Reply:
> this is tricky. as you wrote, if the `-d` is invalid we can search for `-b`. or as `-d` is invalid we can raise error and stop, or show error message and continue.
for the user, although `-d` is invalid and the tool continues, user may not notice there is an error. and this may confuse users. (or some users may think the tool is clever)
for me, I would go with raise error and stop.

_Originally posted by @seanshpark in https://github.com/Samsung/ONE/pull/9175#discussion_r883139663_


Let's go with exclusive relation between two options! 

`-d` | `-b` | log
-- | -- | :-- 
o | o | `Error: -d and -b options are mutually exclusive. Please use only one of them`
o | x | \-
x | o | \-
x | x | `Error: Following arguments are required. {-d/--driver \| -b/--backend}, ...`